BGL/include/CGAL/boost/graph/METIS/partition_dual_graph.h

```diff
@@ -98,7 +98,7 @@ void partition_dual_graph(const TriangleMesh& tm,
   CGAL_assertion_code(int ret =)
     METIS_PartMeshDual(&ne, &nn, eptr, eind,
                        nullptr /* elements weights*/, nullptr /*elements sizes*/,
-                       &ncommon, &nparts,
+                       &ncommon, static_cast<idx_t*>(&nparts),
```

Comment:
That cannot work. The variable `nparts` is a parameter of the function, of type `int`, on the stack. If `idx_t` and `int` have different size (and they seem so), then reading `nparts` through a point to `long long` will add 32 bits to the type, and whatever is in memory in those 32 bits (4 octets) will be added to the value as the 32 most significant bits.

The right solution is to create a real variable of the right type, with the right value:
```c++
idx_t nparts_as_idx_t = nparts;
// [...]
METIS_PartMeshDual(&ne, &nn, eptr, eind,
                       nullptr /* elements weights*/, nullptr /*elements sizes*/,
                       &ncommon, &nparts_as_idx_t,
                       nullptr /* partitions weights */,
                       *options,
                       &objval, epart, npart);
```

Reply:
Right, thank you.